### ISSUE TITLE
IllegalArgumentException: Comparison method violates its general contract at utilTimSort

### ISSUE THREAD

**Issue by Lonami** [00-Evan/shattered-pixel-dungeon#1490]
Seems the TimSort failed. Maybe I overdid it with the spawns…

<details><summary>Screenshots</summary>

![frozen frame](https://github.com/00-Evan/shattered-pixel-dungeon/assets/6297805/dcb18d13-89c8-4a99-bfec-8349b6453a90)

![crash popup](https://github.com/00-Evan/shattered-pixel-dungeon/assets/6297805/3f2f0da0-fbb3-4dba-b816-474ec8008eba)

</details>

<details><summary>OCR-d crash log</summary>

**Shattered Pixel Dungeon Has Crashed!**

Shattered Pixel Dungeon has run into an error It cannot recover from and has crashed, sorry about that!

If you could, please email this error message to the developer (...):

```
version: 2.2.0-BETA-0.1
java.lang.IllegalArgumentException: Comparison method violates its general contract!
at java.base/java.utilTimSort.mergeLo(781)
at java.base/java.utilTimSortmergeAt(518)
at java.base/java.utilTimSort.mergeForceCollapse(461)
at java.base/java.utilTimSort.sort(254)
at java.base/java.utilArrays.sort(1307)
at java.base/java.util.ArrayList.sort(1721)
at java.base/java.util.Collections.sort(179)
at noosa.Group.sort(317)
at scenes.GameScene.sortMobSprites(895)
at sprites.CharSprite.onComplete(727)
at noosa.tweeners.Tweener.onComplete(75)
at noosa.tweeners.Tweener.update(62)
at noosa.Group.update(66)
at noosa.Group.update(66)
at noosa.Scene.update(63)
at scenes.PixelScene.update(199)
at scenes.GameScene.update(693)
at noosa.Game.update(282)
at noosa.Game.step(245)
at noosa.Game.render(175)
at backends.lwjg13.Lwjg13Window.update(387)
at backends.lwjg13.1.wjg13Application.loop(192)
at backends.lwjg13.bvjg13Application.<ini...
```

</details>

I also had an another crash during this run, but the game just froze with no crash popup. Thanks for the great open-source game!

**Comment by 00-Evan** [comment]
This should already be fixed as of BETA-0.3.

regarding the freeze, can you give me any more information about when the freeze occured? I've got some scattered reports about freezes in the beta but nothing conclusive yet.

**Comment by Lonami** [comment]
Thanks!

Sorry, no clue why it happened, and can't remember many details by now. I know I was being chased on low HP and after the freeze, I closed the game. After opening it back up, the enemies that were chasing me were no longer there, and different ones seemed to take place.

Although maybe that's simply due to RNG and the last save-state being at an earlier point, spawning new random mobs.
